Feature: CustomPrintNodeFn should be able to return undefined

Lexical version: 0.31.2

## Steps To Reproduce

1. Use a `TreeView` from `@lexical/react/LexicalTreeView` an supply a custom `customPrintNode` function.
2. (Implicitly) return `undefined` to get the default behaviour for nodes whose representation you don't want to override.
3. Type error due to `CustomPrintNodeFn` return type being `string`.

## The current behavior

TypeScript type-checking fails:

```
error TS2322: Type '(node: LexicalNode) => string | undefined' is not assignable to type 'CustomPrintNodeFn'.
       Type 'string | undefined' is not assignable to type 'string'.
         Type 'undefined' is not assignable to type 'string'.
```

## The expected behavior

Type checking succeeds.

## More info

The code allows returning `undefined` from `customPrintNode` to get the default behaviour:

https://github.com/facebook/lexical/blob/36493cb0efc9d5156185aa8db51223f41b0a9afe/packages/lexical-devtools-core/src/generateContent.ts#L261-L266

--- a/packages/lexical-devtools-core/src/generateContent.ts
+++ b/packages/lexical-devtools-core/src/generateContent.ts
@@ -35,7 +35,7 @@ import {LexicalCommandLog} from './useLexicalCommandsLog';
 export type CustomPrintNodeFn = (
   node: LexicalNode,
   obfuscateText?: boolean,
-) => string;
+) => string | undefined;
 
 const NON_SINGLE_WIDTH_CHARS_REPLACEMENT: Readonly<Record<string, string>> =
   Object.freeze({

